packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/relay/relay.ts

```diff
@@ -46,34 +92,36 @@ export const relay = async (
     const strategy =
       confirmationOptions?.strategy ?? (await connection.getLatestBlockhash())
     const decoded = Buffer.from(encodedTransaction, 'base64')
-    let transaction = VersionedTransaction.deserialize(decoded)
+    const transaction = VersionedTransaction.deserialize(decoded)
 
-    const decompiled = TransactionMessage.decompile(transaction.message)
-    const feePayerKey = transaction.message.getAccountKeys().get(0)
+    const lookupTableAccounts = await getLookupTableAccounts(
+      transaction.message.addressTableLookups.map((k) => k.accountKey)
+    )
+    const decompiled = TransactionMessage.decompile(transaction.message, {
+      addressLookupTableAccounts: lookupTableAccounts
+    })
+
+    const feePayerKey = decompiled.payerKey
     const feePayerKeyPair = getFeePayerKeyPair(feePayerKey)
-    if (!feePayerKey || !feePayerKeyPair) {
-      throw new BadRequestError(
-        `No fee payer for address '${feePayerKey?.toBase58()}'`
-      )
-    }
     await assertRelayAllowedInstructions(decompiled.instructions, {
       user: res.locals.signerUser,
       feePayer: feePayerKey.toBase58()
     })
 
-    if (isPaymentTransaction(decompiled.instructions)) {
```

Comment:
i'll take the AI of having this return a thing to the client for the client to attach to the tx